packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1010,6 +1020,8 @@ export class RemoteFluidDataStoreContext extends FluidDataStoreContext {
 		}
 	}
 
+	public setAttachState(attachState: AttachState.Attaching | AttachState.Attached) {}
```

Comment:
Can we assert that should never be called?

Reply:
It is actually called. I do not remember where, but I broke a UT when working this PR, thus I know.
But I forgot the actual scenario - let me reproduce it.

Overall, the way this code was structured (with events and using once()) was that they were quite often called multiple times.

Reply:
The scenario (hit by stashedOps.spec.ts, "resends attach op" UT is the following (as far as I understand):
1. data store is being attached in attached container
2. container state is serialized
3. container state is rehydrated
    - as result, we create RemoteFluidDataStoreContext for this data store that is actually in "attaching" state (as of # 2). But its state is set to attached when loading container from stashed ops
4. attach op for this data store is processed - setAttachState() is called.

Reply:
Added big comment about it in the code itself.

Reply:
Ah okay. I thought during rehydration we created Rehydrated contexts. Anyway, that makes sense.